Your blog post should include the following:

A compelling title about your findings
An introduction to the data research
A section sharing the background info (definition of GDP for example) and sources for your data as well as any further research you conducted
An accompanying paragraph describing the following visualizations
A conclusion touching on the limitations of the data and further research
Visualizations that can be included:

The violin plot of the life expectancy distribution by country
The facet grid of scatter graphs mapping GDP as a function of Life Expectancy by country
The facet grid of line graphs mapping GDP by country
The facet grid of line graphs mapping Life Expectancy by country

In [28]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns

**Original Data Source - Life Expectancy:** https://apps.who.int/gho/athena/api/GHO/life_expectancy?target=GHO/WHOSIS_000001&profile=text&format=csv   
**Original Data Source - GDP per Capita:** https://api.worldbank.org/v2/en/indicator/NY.GDP.PCAP.CD?downloadformat=csv

In [29]:
life_expectancy = pd.read_csv('life_expectancy.csv')

In [30]:
life_expectancy.head(15)

Indicator PUBLISH STATES  Year  \
0   Life expectancy at birth (years)      Published  2010   
1   Life expectancy at birth (years)      Published  2010   
2   Life expectancy at birth (years)      Published  2010   
3   Life expectancy at birth (years)      Published  2010   
4   Life expectancy at birth (years)      Published  2010   
5   Life expectancy at birth (years)      Published  2010   
6   Life expectancy at birth (years)      Published  2015   
7   Life expectancy at birth (years)      Published  2015   
8   Life expectancy at birth (years)      Published  2015   
9   Life expectancy at birth (years)      Published  2015   
10  Life expectancy at birth (years)      Published  2015   
11  Life expectancy at birth (years)      Published  2015   
12  Life expectancy at birth (years)      Published  2019   
13  Life expectancy at birth (years)      Published  2019   
14  Life expectancy at birth (years)      Published  2019   

               WHO region World Bank income group               Country  \
0   Eastern Mediterranean                     NaN           Afghanistan   
1   Eastern Mediterranean                     NaN           Afghanistan   
2   Eastern Mediterranean                     NaN           Afghanistan   
3                  Africa                     NaN                Angola   
4                  Africa                     NaN                Angola   
5                  Africa                     NaN                Angola   
6                  Africa                     NaN                Angola   
7                  Europe                     NaN               Albania   
8                  Europe                     NaN               Albania   
9                  Europe                     NaN               Albania   
10  Eastern Mediterranean                     NaN  United Arab Emirates   
11  Eastern Mediterranean                     NaN  United Arab Emirates   
12  Eastern Mediterranean                     NaN  United Arab Emirates   
13               Americas                     NaN             Argentina   
14               Americas                     NaN             Argentina   

           Sex  Display Value   Numeric  
0         Male           59.6  59.60036  
1       Female           60.3  60.29720  
2   Both sexes           59.9  59.94055  
3         Male           55.8  55.77834  
4       Female           60.5  60.52821  
5   Both sexes           58.1  58.06943  
6   Both sexes           61.7  61.71907  
7         Male           76.1  76.10220  
8       Female           79.7  79.68682  
9   Both sexes           77.8  77.84428  
10        Male           74.3  74.25413  
11      Female           77.8  77.84315  
12  Both sexes           76.1  76.07952  
13        Male           73.5  73.51080  
14      Female           79.5  79.50281

##### What does each column represent?
* **Indicator:**  (UNNECESARY)  
* **PUBLISH STATES:**  (UNNECESARY)  
* **Year:**  
* **WHO region:**  
* **World Bank income group:**  
* **Country:**  
* **Sex:**  
* **Display Value:**  
* **Numeric:**  (UNNECESARY)   

Lets drop the unnecesary columns in order to simplify further work.

In [31]:
life_expectancy.drop(['Indicator','PUBLISH STATES','Numeric'], axis=1, inplace=True)


In [32]:
life_expectancy.drop_duplicates()
life_expectancy.columns = map(str.lower, life_expectancy.columns)
life_expectancy.info()
life_expectancy.describe(include='all')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2328 entries, 0 to 2327
Data columns (total 6 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   year                     2328 non-null   int64  
 1   who region               2280 non-null   object 
 2   world bank income group  60 non-null     object 
 3   country                  2196 non-null   object 
 4   sex                      2328 non-null   object 
 5   display value            2328 non-null   float64
dtypes: float64(1), int64(1), object(4)
memory usage: 109.2+ KB


year who region world bank income group      country   sex  \
count   2328.000000       2280                      60         2196  2328   
unique          NaN          7                       5          183     3   
top             NaN     Europe              Low-income  Afghanistan  Male   
freq            NaN        612                      12           12   776   
mean    2011.000000        NaN                     NaN          NaN   NaN   
std        7.107862        NaN                     NaN          NaN   NaN   
min     2000.000000        NaN                     NaN          NaN   NaN   
25%     2007.500000        NaN                     NaN          NaN   NaN   
50%     2012.500000        NaN                     NaN          NaN   NaN   
75%     2016.000000        NaN                     NaN          NaN   NaN   
max     2019.000000        NaN                     NaN          NaN   NaN   

        display value  
count     2328.000000  
unique            NaN  
top               NaN  
freq              NaN  
mean        70.233033  
std          8.831624  
min         28.000000  
25%         64.400000  
50%         71.800000  
75%         77.000000  
max         86.900000

In [33]:
print(life_expectancy['who region'].unique())
print(life_expectancy['world bank income group'].unique())
print(life_expectancy['sex'].unique())

['Eastern Mediterranean' 'Africa' 'Europe' 'Americas' 'Western Pacific'
 'South-East Asia' nan 'Global']
[nan 'Low-income' 'Lower-middle-income' 'Upper-middle-income'
 'High-income' 'Global']
['Male' 'Female' 'Both sexes']


## Structurally Missing Data
This dataset contains 2328 measurements of average life expectancy (at birth) from 7 WHO regions ('Eastern Mediterranean', 'Africa', 'Europe', 'Americas', 'Western Pacific',
 'South-East Asia' and 'Global'), 5 World Bank income groups ('Low-income', 'Lower-middle-income', 'Upper-middle-income', 'High-income' and 'Global'), 183 countries and 3 sex divisiones ('Male', 'Female' and 'Both sexes').  
Although it seems there are missing values for 'WHO region', 'World Bank income group' and 'Country' that's only because there are certain rows dedicated exclusively to contain the average life expectancy for a whole 'WHO region' or 'World Bank income group', thus they do not display any value in the 'Country' feature. Alternatively, rows for global life expectancy (12 in total, for each combination of year and sex) contain the value 'global' both in the cells for 'WHO region' or 'World Bank income group'. Thus the irregular number on null values in some columns.

In [36]:
life_expectancy_2000 = life_expectancy[life_expectancy.year == 2000]
life_expectancy_2010 = life_expectancy[life_expectancy.year == 2010]
life_expectancy_2015 = life_expectancy[life_expectancy.year == 2015]
life_expectancy_2019 = life_expectancy[life_expectancy.year == 2019]